### PR TITLE
Create load timeout error and guard trigger error

### PR DESCRIPTION
**Pull request by JustinHoldstock**
Adding better guarding around empty `err` param, or string param, as well as providing a PreviewError upon viewer load timeout

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @JustinHoldstock has signed the CLA. Thanks for the pull request!

File: src/lib/viewers/BaseViewer.js [hunk 1]
```diff
@@ -300,10 +302,17 @@ class BaseViewer extends EventEmitter {
...
-            err instanceof PreviewError
-                ? err
-                : new PreviewError(ERROR_CODE.LOAD_VIEWER, __('error_refresh'), {}, err.message);
+        let error;
+
+        if (err instanceof PreviewError) {
+            error = err;
+        } else {
+            let message = err || '';
+            message = typeof message === 'string' ? message : err.message;
```
Comment (tonyjin) [inline src/lib/viewers/BaseViewer.js]: Do we have any code that triggers an error with just a string?

Comment (JustinHoldstock) [reply src/lib/viewers/BaseViewer.js]: I'm adding this in as a guard until we're 100% certain we aren't allowing `triggerError()` to accept a string. I couldn't find any instances where we do, but there are quite a few places that call triggerError(), and I don't want anything to blow up.

Comment (jeremypress) [reply src/lib/viewers/BaseViewer.js]: I'm fine with punishing a bad use of triggerError here. If you're not passing in a PreviewError then you get a generic error load viewer code. 

If we don't have this usage anywhere right now, we should prevent it before anyone starts doing that.

File: src/lib/viewers/__tests__/BaseViewer-test.js [hunk 2]
```diff
@@ -168,6 +168,22 @@ describe('lib/viewers/BaseViewer', () => {
             // Test cleanup
             clearTimeout(base.loadTimeoutId);
         });
+
+        it('should trigger an error if it takes too long to load', () => {
```
Comment (jeremypress) [inline src/lib/viewers/__tests__/BaseViewer-test.js]: should trigger an error if the viewer times out

File: src/lib/viewers/BaseViewer.js (hunk 1 above)
Comment (JustinHoldstock) [reply src/lib/viewers/BaseViewer.js]: That sounds fine by me! I'll remove the string case from this, and always expect an error to be passed in

**Pull request merged**